src/main/java/com/sandeep/demoemployee/controller/EmployeeController.java

```diff
@@ -0,0 +1,114 @@
+package com.sandeep.demoemployee.controller;
+
+import com.sandeep.demoemployee.entity.CrudeEmployee;
+import com.sandeep.demoemployee.entity.Employee;
+import com.sandeep.demoemployee.entity.NewEmployee;
+import com.sandeep.demoemployee.service.EmployeeService;
+import com.sandeep.demoemployee.util.Messages;
+import io.swagger.annotations.Api;
+import io.swagger.annotations.ApiOperation;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@RestController
+@RequestMapping("/employees")
+@Api
+public class EmployeeController {
+    private final EmployeeService employeeService;
+    private final Messages messages;
+
+    @Autowired
+    public EmployeeController(EmployeeService employeeService, Messages messages) {
+        this.employeeService = employeeService;
+        this.messages = messages;
+    }
+
+    @ApiOperation(value = "View a list of available employees", response = List.class)
+    @GetMapping
+    public ResponseEntity getAllEmployee() {
+        List<Employee> allEmployees = new ArrayList<>(employeeService.getAllEmployees());
+        return new ResponseEntity<>(allEmployees, HttpStatus.OK);
+    }
+
+    @GetMapping("/{id}")
+    public ResponseEntity findAllByEmpId(@PathVariable int id) {
+        if (id < 1) {
+            return new ResponseEntity<>(messages.getMessage("INVALID_ID"), HttpStatus.BAD_REQUEST);
+        }
+        if (employeeService.employeeExists(id))
+            return new ResponseEntity<>(messages.getMessage("EMP_NOT_EXISTS"), HttpStatus.NOT_FOUND);
+        return new ResponseEntity<>(employeeService.generateEmpDetails(id), HttpStatus.OK);
+    }
+
+    @GetMapping("/CountDsgId={id}")                                 //used for debugging
+    public ResponseEntity countByDesignationId(@PathVariable int id) {
+        return new ResponseEntity<>(employeeService.getTotalEmployeeByDesignation(id), HttpStatus.OK);
+    }
+
+    @PostMapping
+    public ResponseEntity addEmployee(@RequestBody CrudeEmployee crudeEmployee) {
+        //crude is needed for designation field
+        if (crudeEmployee.getEmpName() == null || crudeEmployee.getEmpName().matches(".*\\d.*"))
+            return new ResponseEntity<>("", HttpStatus.BAD_REQUEST);
+        Employee employee = employeeService.getEmpFromCrudeEmp(crudeEmployee);
+        if (employeeService.validateEntry(employee))
```

Comment:
ValidEntry() name is confusing, If a employee is valid why it would be BAD_REQUEST.